### PR TITLE
Add Azure to cloud docs

### PR DESCRIPTION
**Pull request by kbatuigas**
## Description

- Add support for Dedicated clusters to Private Link doc
- Add Azure to Cloud API docs

Resolves https://github.com/redpanda-data/documentation-private/issues/2691
Resolves https://github.com/redpanda-data/documentation-private/issues/2692
Review deadline: 28 Aug 2024

## Page previews

[Cloud API Quickstart](https://deploy-preview-38--rp-cloud.netlify.app/redpanda-cloud/manage/api/cloud-api-quickstart/)
[Use the Control Plane API with BYOC](https://deploy-preview-38--rp-cloud.netlify.app/redpanda-cloud/manage/api/cloud-byoc-controlplane-api/)
[Configure Azure Private Link with the Cloud API](https://deploy-preview-38--rp-cloud.netlify.app/redpanda-cloud/networking/azure-private-link/)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Comment by micheleRP** [comment]
@kbatuigas now with Azure in Dedicated, can you please add to the [Dedicated - VPC Peering page](https://deploy-preview-38--rp-cloud.netlify.app/redpanda-cloud/networking/dedicated/vpc-peering/) something about how you can set VPC peering for AWS and GCP deployments (not Azure)?

File: modules/manage/pages/api/cloud-api-quickstart.adoc [hunk 1]
```diff
@@ -19,6 +19,12 @@ BYOC or Dedicated::
 . Create a cluster by making a xref:api:ROOT:cloud-api.adoc#post-/v1beta2/clusters[`POST /v1beta2/clusters`] request.
 . For BYOC, run `rpk cloud byoc`, passing the `metadata.cluster_id` from the Create Cluster response as a flag:
 +
```
Comment (micheleRP) [inline modules/manage/pages/api/cloud-api-quickstart.adoc:21]: @kbatuigas Can you please list these in alphabetical order: AWS, Azure, GCP?

File: modules/manage/partials/controlplane-api.adoc [hunk 2]
```diff
@@ -145,6 +145,13 @@ ifdef::env-byoc[]
 +
```
Comment (micheleRP) [inline modules/manage/partials/controlplane-api.adoc:145]: @kbatuigas Can you please list these in alphabetical order: AWS, Azure, GCP?

**Review by micheleRP — APPROVED** [review]
LGTM, thanks @kbatuigas!

**Review by c4milo — APPROVED** [review]

**Review by bpraseed — APPROVED** [review]
LGTM

**Pull request merged**